rustworkx-core/src/generators/random_graph.rs

```diff
@@ -113,32 +114,46 @@ where
                 }
             }
         } else {
-            let mut v: isize = if directed { 0 } else { 1 };
-            let mut w: isize = -1;
-            let num_nodes: isize = num_nodes as isize;
+            let num_nodes = num_nodes as isize;
```

Comment:
I know this is pre-existing code but couldn't this potentially overflow since the `num_nodes` input is of type `usize`? Should we have a guard against `num_nodes > isize::MAX` or use `try_into()` here?

Reply:
Yeah, `try_into` is a good idea. If it overflows, should we set it to `isize::MAX` or `isize::MIN` or should we error or both?

Reply:
Also, should we say something in the docs about num_nodes not exceeding some value?

Reply:
I think we should probably just error in the case we've exceeded.

We can document it but `isize::MAX` is 9,223,372,036,854,775,807 on 64 bit platforms and ~2 billion on 32 bit platforms, maybe just say it will return an error if the num of nodes exceeds the `isize::MAX` directly (for rustworkx-core). But I think the `InvalidInputError` error struct is probably sufficient to cover this case too.

Reply:
Done in https://github.com/Qiskit/rustworkx/pull/954/commits/fd7544a9e10ecded245780b2771910b8c12794da. I believe this does what you're saying. I wasn't sure how to test this, since anything I tried with very large numbers gave either a `capacity overflow` or `memory allocation` error.

Reply:
Actually I redid this in https://github.com/Qiskit/rustworkx/pull/954/commits/b41cf8a127fb71a91137e2dcfafcfeaaafc9e59d using match. I was able to successfully test locally if I changed all the `isize` to `i8`, but if I test at `isize` levels, I still get capacity or memory errors. So I did not add any new tests for this.